### PR TITLE
Working on Node.js binding

### PR DESCRIPTION
**Pull request by hintjens**
See README.md.

### CODE REVIEW

File: bindings/nodejs/build.sh [hunk 1]
```diff
@@ -122,12 +122,17 @@ cd $BUILD_ROOT
 test ! -d node_modules/nan && npm install nan@latest --save
 test ! -d node_modules/bindings && npm install bindings --save
 
+#   Still not sure of this
```
Comment (rgbkrk) [inline bindings/nodejs/build.sh]: That's a-ok, since we can always build the Electron binaries once the node binaries are in a good state.

**Comment by rgbkrk** [comment]
I'm on OS X here. Went ahead and tried this out, ran into an issue where libs came out of `deps/lib`, though I had it built in `lib/` (while in `bindings/nodejs`):

```
zyre@0.0.1 /Users/kyle6475/code/src/github.com/zeromq/zyre/bindings/nodejs
└── nan@2.2.0

zyre@0.0.1 /Users/kyle6475/code/src/github.com/zeromq/zyre/bindings/nodejs
└── bindings@1.2.1

  CXX(target) Release/obj.target/zyre/binding.o
  SOLINK_MODULE(target) Release/zyre.node
clang: error: no such file or directory: '/Users/kyle6475/code/src/github.com/zeromq/zyre/bindings/nodejs/deps/lib/libczmq.a'
clang: error: no such file or directory: '/Users/kyle6475/code/src/github.com/zeromq/zyre/bindings/nodejs/deps/lib/libzmq.a'
clang: error: no such file or directory: '/Users/kyle6475/code/src/github.com/zeromq/zyre/bindings/nodejs/deps/lib/libsodium.a'
make: *** [Release/zyre.node] Error 1
```

I just moved these manually to see if it would work out. Had to `brew install ossp-uuid` for libuuid then also make sure to `npm install` for deps (though the build itself fails). Run `./build.sh` again then run `node test.js`:

```
$ node test.js
I: 16-02-08 21:03:46 zbeacon: API command=CONFIGURE
I: 16-02-08 21:03:46 zbeacon: interface=en0 address=192.168.0.20 broadcast=192.168.0.255
I: 16-02-08 21:03:46 zbeacon: configured, hostname=192.168.0.20
I: 16-02-08 21:03:46 zbeacon: API command=PUBLISH
I: 16-02-08 21:03:46 (World) JOIN group=GLOBAL
I: 16-02-08 21:03:46 zbeacon: API command=SUBSCRIBE
Hello: World
Waiting for 2 seconds, then exiting
I: 16-02-08 21:03:48 zyre_node: dump state
I: 16-02-08 21:03:48  - name=World uuid=306ED29569EB492DBEC0C052FA0CD631
I: 16-02-08 21:03:48  - endpoint=tcp://192.168.0.20:49152
I: 16-02-08 21:03:48  - discovery=beacon port=5670 interval=0
I: 16-02-08 21:03:48  - headers=0:
I: 16-02-08 21:03:48  - peers=0:
I: 16-02-08 21:03:48  - groups=1:
I: 16-02-08 21:03:48    - GLOBAL
I: 16-02-08 21:03:48 zbeacon: API command=PUBLISH
I: 16-02-08 21:03:48 zbeacon: API command=$TERM
```

**Comment by hintjens** [comment]
So on OS X you're not using autotools, but something else? I'll have to adapt build.sh for that, and also write a build.bat for windows that does the appropriate msbuild magic.

The alternative is to do all C/C++ compiles via node-gyp, which is doable if we use tweetnacl instead of libsodium.

**Comment by rgbkrk** [comment]
Well, I have autoconf and automake (installed via `brew`). Mileage definitely varies among developers on OS X, though sane ones rely on `brew` instead of Apple's provided binaries.

Perhaps I missed a step in the README

**Comment by hintjens** [comment]
Did you run build.sh? It uses --prefix to install into dev/libs. I had
tried using build/libs but `node-gyp rebuild` wipes that directory so the
dependencies have to go elsewhere.

If you need to, do a clean build using `build.sh -f`.

**Comment by rgbkrk** [comment]
Yeah I used build.sh

**Comment by hintjens** [comment]
Well, if you can reproduce it not installing into deps, that'd be helpful.

`./build.sh -f -v` does a verbose full build, good for debugging problems.

**Pull request merged**